lib/http/node.js

```diff
@@ -123,8 +123,14 @@ function getProxyAgent(proxyOptions, parsedUrl, destination, mock) {
       return mockAgent;
     }
   }
-
-  const agentId = createAgentId(agentOptions.protocol, agentOptions.hostname, agentOptions.keepAlive);
+  let destHost;
+  try {
+    destHost = new URL(destination).hostname;
+  } catch (err) {
+    destHost = destination;
```

Comment:
should we log the error here?

Reply:
I added it: https://github.com/snowflakedb/snowflake-connector-nodejs/pull/946/files#diff-3fb720f70ea34fb2975fd6c38ac4b6b6131373be2abfa4760dfaae8a25daaee2R905